pulsar-broker/src/main/java/org/apache/pulsar/PulsarBrokerStarter.java

```diff
@@ -56,12 +138,62 @@ public static void main(String[] args) throws Exception {
         String configFile = args[0];
         ServiceConfiguration config = loadConfig(configFile);
 
+        CommandLine cmdLine = readCommandLine(args);
+        boolean runBookie = runBookieTogether(cmdLine);
+        boolean runBookieAutoRecovery = runBookieAutoRecoveryTogether(cmdLine);
+        final BookieServer bookieServer;
```

Comment:
I would suggest wrapping `BookieServer` `AutoRecoveryMain` and `StatsProvider` into a static inner class called `PulsarBookieStarter` and provide a method to `start` and a method to `stop`. so it is easier for unit testing. You can unit test `PulsarBookieStarter` as well.

Reply:
Thanks, will change it.